packages/scratch-vm/docs/typescript.md

```diff
@@ -145,4 +145,79 @@ class BarebonesTs extends Extension<{
 }
 
 export = BarebonesTs;
-```
\ No newline at end of file
+```
+
+## How to declare menu details
+
+### Allowed
+
+```ts
+class GoodInline extends Extension<{
+  title: "my_title",
+  description: "my_description",
+  iconURL: "my_iconURL",
+  insetIconURL: "my_insetIconURL",
+}, {}>{
+  ...
+}
+
+```
+
+### Allowed
+
+```ts
+type DisplayDetails = {
+  title: "my_title",
+  description: "my_description",
+  iconURL: "my_iconURL",
+  insetIconURL: "my_insetIconURL",
+}
+
+class GoodInSameFile extends Extension<DisplayDetails, {}>{
+  init = notImplemented;
+  blockBuilders = notImplemented;
+}
+```
+
+### NOT Allowed
+
+```ts
+type Title = "my_title";
+type Description = "my_description";
+type IconURL = "my_iconURL";
+type InsetIconURL = "my_insetIconURL";
+
+class BadInline extends Extension<{
+  title: Title,
+  description: Description,
+  iconURL: IconURL,
+  insetIconURL: InsetIconURL,
+}, {}>{
+  ...
+}
+```
+
+### NOT Allowed
+
+```ts
+type Title = "test_title";
+type Description = "test_description";
+type IconURL = "test_iconURL";
+type InsetIconURL = "test_insetIconURL";
+
+type DisplayDetails = {
+  title: Title,
+  description: Description,
+  iconURL: IconURL,
+  insetIconURL: InsetIconURL,
+}
+
+class BadSameFile extends Extension<DisplayDetails, {}>{
+  ...
+}
+```
+
```

Comment:
Add not allowed for interpolated strings:

type DisplayDetails = {
  title: `${}_hello`,
  description: Description,
  iconURL: IconURL,
  insetIconURL: InsetIconURL,
}

Reply:
^This is for @pmalacho-mit